flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/metrics/utils/MetricsPredicateProvider.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.metrics.utils;
+
+import org.apache.flink.runtime.execution.ExecutionState;
+import org.apache.flink.runtime.jobgraph.JobType;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+/** Helper class which provides relevant predicates for state metrics. */
+public class MetricsPredicateProvider {
```

Comment:
This class doesn't scale well into the future; it will quickly become unreadable should more states be added.

Consider creating some static predicates, with each only caring about a single state. Then you also don't have to worry about incorrect target states.